Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
Okay, I ran this test.  There are flaws in the testing approach.

First, and critical:
SendMessage("SendMessage") should be changed to SendMessage("SendAMessage") or something else that doesn't resolve to something that requires parameters.  Most of the time in SendMessage as the script is written is spent in a Debug.LogError that SendMessage requires three parameters.  

After this:
![image](https://cloud.githubusercontent.com/assets/15933792/18889251/093be486-84b1-11e6-96e7-e740a99f5394.png)


                



Second:
MaxListeners should not be 1.  It should be set to some number or UI controls that might be in the scene.  Maybe 20 is a more reasonable starting point (The shell has 13 buttons on the start page.  Add a few windows/holograms to the world and 20 is not an excessive number.  My personal app has closer to 50 individual buttons in the world).  I set the MaxListeners to 20. Now this adds pressure to how long the script takes to run (it seems to hang actually), so I decreased numreps to 100000.  Now I get:

![image](https://cloud.githubusercontent.com/assets/15933792/18889383/9774e752-84b1-11e6-9f32-ebd2accf6685.png)